crates/stream-download/src/storage/mod.rs

```diff
@@ -9,6 +9,142 @@ pub mod memory;
 #[cfg(feature = "temp-storage")]
 pub mod temp;
 
+/// Represents a content length that can change during processing.
+///
+/// This structure holds:
+/// - `reported`: The length of the content as reported by the server (e.g., Content-Range header).
+/// - `gathered`: The actual length of the content after processing (e.g., decryption,
+///   decompression), or `None` if the final value is not yet known (still being calculated or
+///   accumulated).
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub struct DynamicLength {
+    /// The length of the content as reported by the server, for example, via Content-Range header.
+    pub reported: u64,
```

Comment:
I don't think we want to allow consumers to modify these values. These should probably be kept private.